lib/logstash/filters/geoip.rb

```diff
@@ -166,8 +170,12 @@ def filter(event)
       # Dont' swallow this, bubble up for unknown issue
       raise e
     end
-
-    event[@target] = geo_data_hash
+    if @merge
+      event[@target] ||= {}
+      event[@target].merge!(geo_data_hash)
```

Comment:
`.merge!` is not a safe thing to do on a Logstash Event. You should insert each key separately.